### PR TITLE
Better step input descriptions

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [X] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

Update `step.yml` with up to date input descriptions. Also update the overall step description.

### CODE REVIEW

**Comment by ofalvai** [comment]
Current dependencies on/for this PR:
* next
  * **PR #195** <a href="https://app.graphite.dev/github/pr/bitrise-steplib/steps-git-clone/195" target="_blank">Graphite</a> 
    * **PR #196** <a href="https://app.graphite.dev/github/pr/bitrise-steplib/steps-git-clone/196" target="_blank">Graphite</a> 
      * **PR #197** <a href="https://app.graphite.dev/github/pr/bitrise-steplib/steps-git-clone/197" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/bitrise-steplib/steps-git-clone/197?utm_source=stack-comment).

File: README.md [hunk 1]
```diff
@@ -2,28 +2,27 @@
...
-Optionally, you can modify the following fields in the **Clone Config** section:
-1. You can set the **Update the registered submodules?** option to `yes` to pull the most up-to-date version of the submodule from the submodule's repository.
-2. You can set the number of commits you want the Step to fetch in the **Limit fetching to the specified number of commits** option. Make sure you set a decimal number.
+### Configuring the Step
 
-Other **Clone config** inputs are not editable unless you go to the **bitrise.yml** tab, however, to avoid issues, we suggest you to contact our Support team instead.
+The step should work with its default configuration if build triggers and webhooks are set up correctly.
 
-### Troubleshooting
-If you have GitHub Enterprise set up, it works slightly differently on [bitrise.io](https://www.bitrise.io) than on [github.com](https://github.com). You have to manually set the git clone URL, register the SSH key and the webhook.
```
Comment (ofalvai) [inline README.md:24]: This is linked below in the useful links section.

File: README.md [hunk 2]
```diff
@@ -2,28 +2,27 @@
...
-If you face network issues in the case of self-hosted git servers, we advise you to contact our Support Team to help you out.
```
Comment (ofalvai) [inline README.md:25]: This is precisely the thing that our support can't help with 😭

Comment (hisaac) [reply README.md:25]: 😅

**Review by hisaac — APPROVED** [review]

**Pull request merged**